src/io.c

```diff
@@ -83,6 +88,14 @@ static INLINE int TranslateReturnCode(int old, int sd)
         if (errno == RTCSERR_TCP_TIMED_OUT)
             errno = SOCKET_EAGAIN;
     }
+#elif defined( FREERTOS_TCP )
+	/* When using 'ipconfigSUPPORT_SIGNALS', the TCP API's can get interrupted
+	by calling 'FreeRTOS_SignalSocket()' or 'FreeRTOS_SignalSocketFromISR()'.
+	The interrupted API call will return '-pdFREERTOS_ERRNO_EINTR'. */
+	if( old == -pdFREERTOS_ERRNO_EINTR ) {
+		freertos_errno = SOCKET_EINTR;
```

Comment:
Can we just use the system errno here, and set "errno = SOCKET_EINTR" to avoid the extra freertos_errno variable?  Then return -1, also.  We reset errno here in the case of MQX similarly.